healthcheck.go

```diff
@@ -53,6 +54,52 @@ type ClusterHealth struct {
 	Status string `json:"status"`
 }
 
+//indexcheck calls elasticsearch to check if the required indexes from the client exist
+func (cli *Client) indexcheck(ctx context.Context) (code int, err error) {
+
+	for _, index := range cli.indexes {
+
+		urlIndex := cli.url + "/" + index
+		logData := log.Data{"url": urlIndex, "index": index}
+
+		_, err := url.Parse(urlIndex)
+		if err != nil {
+			log.Event(ctx, "failed to create url for elasticsearch indexcheck", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		req, err := http.NewRequest("HEAD", urlIndex, nil)
+		if err != nil {
+			log.Event(ctx, "failed to create request for indexcheck call to elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+
+		if cli.signRequests {
+			awsauth.Sign(req)
+		}
+
+		resp, err := cli.httpCli.Do(ctx, req)
+		if err != nil {
+			log.Event(ctx, "failed to call elasticsearch", log.ERROR, logData, log.Error(err))
+			return 500, err
+		}
+		defer resp.Body.Close()
+		logData["http_code for"] = resp.StatusCode
```

Comment:
This will be invalid JSON output in our logger - either remove `for` or add an `_` 👍